### PR TITLE
feat: add comments and pause for governor

### PR DESCRIPTION
**Pull request by cosinlink**
### Description
1. add pause method for BSCGovernor
2. add comments for BSCGovernor and GovToken

### CODE REVIEW

File: contracts/BC_fusion/BSCGovernor.sol [hunk 1]
```diff
@@ -41,12 +41,30 @@ contract BSCGovernor is
     /*----------------- errors -----------------*/
     error NotWhitelisted();
     error TotalSupplyNotEnough();
+    error GovernorPaused();
```
Comment (zjubfd) [inline contracts/BC_fusion/BSCGovernor.sol:44]: typo, `nor`==> `not`

File: contracts/BC_fusion/BSCGovernor.sol [hunk 2]
```diff
@@ -41,12 +41,30 @@ contract BSCGovernor is
...
+    /*----------------- events -----------------*/
+    event Paused();
+    event Resumed();
 
     /*----------------- storage -----------------*/
     // target contract => is whitelisted for governance
     mapping(address => bool) public whitelistTargets;
-
     bool public proposeStarted;
+    bool private _paused;
```
Comment (zjubfd) [inline contracts/BC_fusion/BSCGovernor.sol:55]: why private? I would like to know whether this contract is paused or not now.

File: contracts/BC_fusion/BSCGovernor.sol [hunk 3]
```diff
@@ -194,7 +267,7 @@ contract BSCGovernor is
         uint256[] memory values,
         bytes[] memory calldatas,
         bytes32 descriptionHash
-    ) internal override(GovernorUpgradeable, GovernorTimelockControlUpgradeable) {
+    ) internal override(GovernorUpgradeable, GovernorTimelockControlUpgradeable) whenNotPaused {
```
Comment (buddh0) [inline contracts/BC_fusion/BSCGovernor.sol:270]: why whenNotPaused is only add for a view function?

File: contracts/BC_fusion/BSCGovernor.sol (hunk 2 above)
Comment (cosinlink) [reply contracts/BC_fusion/BSCGovernor.sol]: modified to public

File: contracts/BC_fusion/BSCGovernor.sol (hunk 3 above)
Comment (cosinlink) [reply contracts/BC_fusion/BSCGovernor.sol:270]: it is not a view function

File: contracts/BC_fusion/BSCGovernor.sol [hunk 4]
```diff
@@ -73,9 +91,34 @@ contract BSCGovernor is
         whitelistTargets[GOVERNOR_ADDR] = true;
         whitelistTargets[GOV_TOKEN_ADDR] = true;
         whitelistTargets[TIMELOCK_ADDR] = true;
+
+        governorProtector = 0xdF87F0e2B8519Ea2DD4aBd8B639cdD628497eD25; // TODO
```
[the remark this answers is not in the document]
Comment (cosinlink) [reply contracts/BC_fusion/BSCGovernor.sol]: fixed

**Pull request merged**